core_lib/session/user_security.py

```diff
@@ -37,7 +38,11 @@ def token_to_session_object(self, token):
 
     def _secure_entry(self, request, policies):
         session_obj = None
-        if self.cookie_name in request.COOKIES:
-            token = request.COOKIES[self.cookie_name]
+        if WebHelpersUtils.get_server_type() == WebHelpersUtils.ServerType.DJANGO:
+            if self.cookie_name in request.COOKIES:
+                token = request.COOKIES[self.cookie_name]
+                session_obj = self.from_session_data(self.token_handler.decode(token))
+        elif request.cookies.get(self.cookie_name):
+            token = request.cookies.get(self.cookie_name)
             session_obj = self.from_session_data(self.token_handler.decode(token))
         return self.secure_entry(request, session_obj, policies)

```

Comment:
@shubham-shay-te do we need to do a `if token:`

Reply:
`if self.cookie_name in request.COOKIES:` and `elif request.cookies.get(self.cookie_name):` are checking the same thing

Reply:
maybe, but on line `48` we are referring to token. while there is no `else` what is it's not Django and cookie not set? token is None. and we try to decode None?

Reply:
yeah if it is not Django and any other framework it will go inside that is why we have kept this inside the condition that if server type is flask then we will check cookies like this. Thank you

Reply:
got it, yeah we might need to add a `if` in this case

Reply:
returned None in the else condition if the token is not there

Reply:
this will not help for the reason that you have another if statment inside the if in line 41, 42. 
please add if to check if the token is there